snapcraft/tests/test_sources.py

```diff
@@ -68,6 +71,89 @@ def test_pull_tarball_must_download_to_sourcedir(self, mock_prov):
         with open(os.path.join(dest_dir, tar_file_name), 'r') as tar_file:
             self.assertEqual('Test fake compressed file', tar_file.read())
 
+    def test_checksum(self):
```

Comment:
@sergiusens this function uses the various supported checksum formats and checks the checksum with them. This ensures that the checksum can be successfully checked with the raw checksums.

The only reason why I had to split it into separate functions was because the static tests were complaining about check_checksum being too complex.